From acb816be7577f4a039a7dffd2b29a3dc627fa70c Mon Sep 17 00:00:00 2001
From: Marco Martin <notmart@gmail.com>
Date: Thu, 28 Sep 2023 16:10:28 +0200
Subject: [PATCH] Only preload a couple of screens worth of comments

Discover used to preload every comment in order to have the scrollbar
behaving as good as possible with variable height items.
However some applications like Firefox have thousands of comments
which will freeze firefox for minutes when scrolling quickly,
eat a lot of memory and will make discover even slow to quit
in order to delete the thousands of items it created.

Cache only a couple of screens worth of comments, so that applications with few comments
behave well and applications with thousands remain efficient

BUG:474944
---
 discover/qml/ReviewsPage.qml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/discover/qml/ReviewsPage.qml b/discover/qml/ReviewsPage.qml
index 5e157885e..89477deae 100644
--- a/discover/qml/ReviewsPage.qml
+++ b/discover/qml/ReviewsPage.qml
@@ -74,7 +74,8 @@ Kirigami.OverlaySheet {
         bottomMargin: Kirigami.Units.largeSpacing
         spacing: Kirigami.Units.smallSpacing
         implicitWidth: Kirigami.Units.gridUnit * 25
-        cacheBuffer: Math.max(0, contentHeight)
+        // Still preload some items to make the scrollbar behave better, but can't preload all the comments as some apps like Firefox have thousands of them which will freeze Discover for minutes
+        cacheBuffer: height * 2
         reuseItems: true
 
         delegate: ReviewDelegate {
-- 
GitLab

